python_modules/dagster/dagster/_core/definitions/metadata/__init__.py

```diff
@@ -1156,26 +1156,40 @@ class NamespacedMetadataEntries(ABC, BaseModel, frozen=True):
     def namespace(cls) -> str:
         raise NotImplementedError()
 
+    @classmethod
+    def _namespaced_key(cls, key: str) -> str:
+        return f"{cls.namespace()}/{key}"
+
+    @staticmethod
+    def _strip_namespace_from_key(key: str) -> str:
+        return key.split("/", 1)[1]
+
+    def dict(self):
+        return {self._namespaced_key(key): getattr(self, key) for key in self.__fields__.keys()}
+
```

Comment:
is this really necessary? How does this different from the default pydantic impl?

Reply:
Pydantic’s dict() JSONizes all the way through. This means that the value of type TableSchema gets converted to a vanilla tuple. Vanilla tuples are not currently accepted as values in the dicts supplied to ‘metadata’ arguments.

A couple thoughts here:
- Maybe this means we shouldn’t use / override dict. However, if we don’t override it I think it creates a footgun if someone stumbles across it. We could also just not use pydantic at all.
- Longer term it might make sense to move towards a world where metadata values are expected to be JSON blobs instead of instances of typed classes.

Reply:
> We could also just not use pydantic at all.

Yeah if we are overriding behavior that is that core to pydantic I lean towards not using it.

However, if convert all the metadata values to pydantic then we are back at expected behavior. I believe we can do that safely now that `serdes` supports pydantic models. This would address your second point around json blobs I believe, and is probably the cleanest solution.

Reply:
Took this out based on our conversation and the discovery of https://lyz-code.github.io/blue-book/coding/python/pydantic_exporting/